frontend/fingertips-frontend/components/organisms/PopulationPyramidWithTable/index.tsx

```diff
@@ -2,65 +2,69 @@
 import { TabContainer } from '@/components/layouts/tabContainer';
+import { H3, H5 } from 'govuk-react';
 import {
-  AreaSelectInputData,
-  AreaSelectInputField,
-} from '@/components/molecules/SelectInputField';
-import { HeaderChartTitle } from './HeaderChartTitle';
-import { H3 } from 'govuk-react';
-import { getLatestYear } from '@/lib/chartHelpers/chartHelpers';
-import {
-  SearchParams,
-  SearchStateManager,
-  SearchStateParams,
-} from '@/lib/searchStateManager';
-import { usePathname, useRouter } from 'next/navigation';
+  determineHealthDataForArea,
+  seriesDataWithoutGroup,
+  sortHealthDataPointsByDescendingYear,
+} from '@/lib/chartHelpers/chartHelpers';
+import { SearchParams, SearchStateParams } from '@/lib/searchStateManager';
 import { ArrowExpander } from '@/components/molecules/ArrowExpander';
 import { PopulationPyramidChartTable } from '../PopulationPyramidChartTable';
 import { DataSource } from '@/components/atoms/DataSource/DataSource';
+import { ChartSelectArea } from '@/components/molecules/ChartSelectArea';
+import { AreaWithoutAreaType } from '../Inequalities/inequalitiesHelpers';
+import {
+  allAreaTypes,
+  HierarchyNameTypes,
+} from '@/lib/areaFilterHelpers/areaType';
+import { areaCodeForEngland } from '@/lib/chartHelpers/constants';
 
-const getHeaderTitle = (
-  healthData: HealthDataForArea | undefined,
-  year: number | undefined,
-  indicatorID: number | undefined
+const determineHeaderTitle = (
+  healthDataForAreaSelected?: HealthDataForArea,
+  areaTypeSelected?: string
 ): string => {
-  let title = undefined;
-  const titleTypeText =
-    indicatorID == nhsIndicatorIdForPopulation ? 'Registered' : 'Resident';
-  if (!year) {
-    title = `${titleTypeText} population profile for ${healthData?.areaName}`;
-  } else {
-    title = `${titleTypeText} population profile for ${healthData?.areaName} ${year}`;
+  if (healthDataForAreaSelected) {
+    const hierarchyName = allAreaTypes.find(
+      (areaType) => areaType.key === areaTypeSelected
+    );
+
+    const titleTypeText =
+      hierarchyName?.hierarchyName === HierarchyNameTypes.NHS
+        ? 'Registered'
+        : 'Resident';
+
+    const year = sortHealthDataPointsByDescendingYear(
+      healthDataForAreaSelected.healthData
+    )[0].year;
+
+    return `${titleTypeText} population profile for ${healthDataForAreaSelected?.areaName} ${year}`;
   }
-  return title;
+  return '';
 };
 
-const getSelectedAreaFrom = (
-  areaCode: string | undefined,
-  convertedData: (PopulationDataForArea | undefined)[]
-) => {
-  let populationArea: PopulationDataForArea | undefined = undefined;
-  if (areaCode && convertedData.length > 0) {
-    populationArea = convertedData.find(
-      (area: PopulationDataForArea | undefined) => area?.areaCode === areaCode
+export function determinePopulationDataForArea(
+  populationDataForAllAreas: PopulationDataForArea[],
+  areaToFind?: string
+) {
+  if (areaToFind) {
```

Comment:
```suggestion
  if (!areaToFind) return populationDataForAllAreas[0];
```
return as early as possible and indent less